### ISSUE TITLE
Unable to upgrade from TM 1.28.5 to TM 1.29.1 - Error:  The Compose file './docker-compose.yml' is invalid

### ISSUE THREAD

**Issue by SemoTech** [teslamate-org/teslamate#3898]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When issuing the "docker-compose pull" command an error about the docker-compose file is thrown and the process fails!

### Expected Behavior

New version should be pulled and the containers updated with no data loss.

### Steps To Reproduce

Issue:
`docker-compose pull`

Result:
```
ERROR: The Compose file './docker-compose.yml' is invalid because:
Unsupported config option for services: 'proxy'
Unsupported config option for volumes: 'teslamate-grafana-data'
```

### Relevant log output

```shell
Here is the current docker-compose.yml file for reference:

[root@teslamate ~]# cat docker-compose.yml 
#version: '3'

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    depends_on:
      - database
    environment:
      - ENCRYPTION_KEY=${TM_ENCRYPTION_KEY} 
      - DATABASE_USER=${TM_DB_USER}
      - DATABASE_PASS=${TM_DB_PASS}
      - DATABASE_NAME=${TM_DB_NAME}
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
      - VIRTUAL_HOST=${FQDN_TM}
      - CHECK_ORIGIN=true
      - TZ=${TM_TZ}
    volumes:
      - ./import:/opt/app/import
    labels:
      - "traefik.enable=true"
      - "traefik.port=4000"
      - "traefik.http.middlewares.redirect.redirectscheme.scheme=https"
      - "traefik.http.middlewares.auth.basicauth.usersfile=/auth/.htpasswd"
      - "traefik.http.routers.teslamate-insecure.rule=Host(`${FQDN_TM}`)"
      - "traefik.http.routers.teslamate-insecure.middlewares=redirect"
      - "traefik.http.routers.teslamate-ws.rule=Host(`${FQDN_TM}`) && Path(`/live/websocket`)"
      - "traefik.http.routers.teslamate-ws.entrypoints=websecure"
      - "traefik.http.routers.teslamate-ws.tls"
      - "traefik.http.routers.teslamate.rule=Host(`${FQDN_TM}`)"
      - "traefik.http.routers.teslamate.middlewares=auth"
      - "traefik.http.routers.teslamate.entrypoints=websecure"
      - "traefik.http.routers.teslamate.tls.certresolver=tmhttpchallenge"
    cap_drop:
      - all

  database:
    image: postgres:12
    restart: always
    environment:
      - POSTGRES_USER=${TM_DB_USER}
      - POSTGRES_PASSWORD=${TM_DB_PASS}
      - POSTGRES_DB=${TM_DB_NAME}
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=${TM_DB_USER}
      - DATABASE_PASS=${TM_DB_PASS}
      - DATABASE_NAME=${TM_DB_NAME}
      - DATABASE_HOST=database
      - GRAFANA_PASSWD=${GRAFANA_PW}
      - GF_SECURITY_ADMIN_USER=${GRAFANA_USER}
      - GF_SECURITY_ADMIN_PASSWORD=${GRAFANA_PW}
      - GF_AUTH_BASIC_ENABLED=true
      - GF_AUTH_ANONYMOUS_ENABLED=false
      - GF_SERVER_ROOT_URL=https://${FQDN_GRAFANA}
      - GF_SMTP_ENABLED=true
      - GF_SMTP_FROM_NAME=${GF_SMTPNAME}
      - GF_SMTP_FROM_ADDRESS=${GF_SMTPSENDER}
      - GF_SMTP_HOST=${GF_SMTPHOST}                      
      - GF_SMTP_USER=${GF_SMTPUSER}                   
      - GF_SMTP_PASSWORD=${GF_SMTPPASS}
      - GF_SMTP_SKIP_VERIFY=false
    volumes:
      - teslamate-grafana-data:/var/lib/grafana
    labels:
      - 'traefik.enable=true'
      - 'traefik.port=3000'
      - "traefik.http.middlewares.redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.grafana-insecure.rule=Host(`${FQDN_GRAFANA}`)"
      - "traefik.http.routers.grafana-insecure.middlewares=redirect"
      - "traefik.http.routers.grafana.rule=Host(`${FQDN_GRAFANA}`)"
      - "traefik.http.routers.grafana.entrypoints=websecure"
      - "traefik.http.routers.grafana.tls.certresolver=tmhttpchallenge"

  mosquitto:
    image: eclipse-mosquitto:1.6
    restart: always
    ports:
      - 127.0.0.1:1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

  proxy:
    image: traefik:v2.1
    restart: always
    command:
      - "--global.sendAnonymousUsage=false"
      - "--providers.docker"
      - "--providers.docker.exposedByDefault=false"
      - "--entrypoints.web.address=:80"
      - "--entrypoints.websecure.address=:443"
      - "--certificatesresolvers.tmhttpchallenge.acme.httpchallenge=true"
      - "--certificatesresolvers.tmhttpchallenge.acme.httpchallenge.entrypoint=web"
      - "--certificatesresolvers.tmhttpchallenge.acme.email=${LETSENCRYPT_EMAIL}"
      - "--certificatesresolvers.tmhttpchallenge.acme.storage=/etc/acme/acme.json"
    ports:
      - 80:80
      - 443:443
    volumes:
      - ./.htpasswd:/auth/.htpasswd
      - ./acme/:/etc/acme/
      - /var/run/docker.sock:/var/run/docker.sock:ro

volumes:
    teslamate-db:
    teslamate-grafana-data:
    mosquitto-conf:
    mosquitto-data:
```

### Type of installation

Docker

### Version

v1.28.5

**Comment by accolon** [comment]
You are still using Docker Compose v1 ("docker-compose") which is deprecated since July 2023. I've seen similar problems to yours because of this old version.

You should migrate to Compose v2 ("docker compose"): https://docs.docker.com/compose/migrate/

**Comment by JakobLichterfeld** [comment]
Also covered in doc: https://docs.teslamate.org/docs/upgrading

**Comment by JakobLichterfeld** [comment]
Additionally your Postgres is outdated, please do a full backup and follow https://docs.teslamate.org/docs/maintenance/upgrading_postgres

**Comment by SemoTech** [comment]
Thanks @accolon and @JakobLichterfeld .  
Issuing "_docker compose pull_" and "_docker compose up -d_" seems to have worked without any more errors!

I am now running TM v1.29.1 since I used docker compose V2 instead of docker-compose V1.  The interface/updating process of the containers is new and more "graphical" so that's nice.  Lastly, I was running out of RAM on the VM when I did "_docker-compose pull_" forcing me to manually pull each component one at the time, but with docker compose V2 that seems to not be an issue as I think it is more RAM optimized and uses less resources.

![image](https://github.com/teslamate-org/teslamate/assets/1452100/9f682237-6d58-46be-9e49-cf8c6d5cbe9c)
As for Postgres, I tried to do the update but the entire system broke afterwards, so I hope I can still run with the older version for a while longer...
